### DISCUSSION TITLE
Help with inline css

### DISCUSSION THREAD

**Discussion by spapas** [django-components/django-components#869] [Q&A] [answered]
Hello friends, I've defined the following simple component to dispaly a bootstrap 5 tooltip:

```python

from typing import Dict, Any
from django_components import Component, register, types


@register("info")
class Info(Component):
    def get_context_data(self, placement: str = "bottom") -> Dict[str, Any]:
        return {
            "placement": placement,
        }

    template: types.django_html = """
<button style='padding-top: 0; padding-bottom: 0; padding-right: 0; margin: 0; border: 0;' class='btn btn-outline-sm' data-bs-toggle="tooltip"  title="{% slot "content" default / %}" data-bs-html="true"  data-bs-placement='{{ placement }}'>
  <svg xmlns="http://www.w3.org/2000/svg" width="20" height="20" fill="currentColor" class="bi bi-info-circle-fill" viewBox="0 0 16 16">
    <path d="M8 16A8 8 0 1 0 8 0a8 8 0 0 0 0 16m.93-9.412-1 4.705c-.07.34.029.533.304.533.194 0 .487-.07.686-.246l-.088.416c-.287.346-.92.598-1.465.598-.703 0-1.002-.422-.808-1.319l.738-3.468c.064-.293.006-.399-.287-.47l-.451-.081.082-.381 2.29-.287zM8 5.5a1 1 0 1 1 0-2 1 1 0 0 1 0 2"></path>
  </svg>
</button>
"""

    css: types.css = """
    .tooltip .tooltip-inner {
        max-width: 400px !important; /* Make the tooltip wider */
    }
    """
```

Notice that this also includes some custom css to make the component wider. However this css isn't rendered anywhere.  Here's what I get in my template:

```html


<!-- _RENDERED Info_4f7345,RLajP1 -->
<button style='padding-top: 0; padding-bottom: 0; padding-right: 0; margin: 0; border: 0;' class='btn btn-outline-sm' data-bs-toggle="tooltip"  title="
              <p>Επιλέξτε τα απογραφικά που θέλετε να στείλετε χρησιμοποιώντας τα φίλτρα. Πρέπει αρχικά να επιλέξετε «Φίλτρο»
                και στη συνέχεια «Μαζική αποστολή». Σε περίπτωση που θέλετε να στείλετε μόνο συγκεκριμένα απογραφικά
              επιλέξτε τα με το πράσινο κουμπί «+» δίπλα στον κωδικό του απογραφικού.</p>
              
            " data-bs-html="true"  data-bs-placement='bottom'>
  <svg xmlns="http://www.w3.org/2000/svg" width="20" height="20" fill="currentColor" class="bi bi-info-circle-fill" viewBox="0 0 16 16">
    <path d="M8 16A8 8 0 1 0 8 0a8 8 0 0 0 0 16m.93-9.412-1 4.705c-.07.34.029.533.304.533.194 0 .487-.07.686-.246l-.088.416c-.287.346-.92.598-1.465.598-.703 0-1.002-.422-.808-1.319l.738-3.468c.064-.293.006-.399-.287-.47l-.451-.081.082-.381 2.29-.287zM8 5.5a1 1 0 1 1 0-2 1 1 0 0 1 0 2"></path>
  </svg>
</button>

```

Can somebody clarify why the css isn't rendered anywhere ?

Also I'd like to know if there's a way to pass context variables to the css, i.e allow something like `        max-width: {{ max_with }} !important; ` for the css of the component.

TIA

**Comment by JuroOravec** [comment, marked as answer]
Hey @spapas, hope you had great holidays :)

----

> Can somebody clarify why the css isn't rendered anywhere ?

How do you render the `Info` component? Do you do `Info.render()`, or insert it into a template with `{% component "info" %}`?

Are you trying to render this as an HTML fragment, or embedded into a proper HTML document with `<head>` and `<body>`?

Since the output contains `<!-- _RENDERED Info_4f7345,RLajP1 -->`, my guess is that the output is not post-processed. [Read more on this in the docs](https://emilstenstrom.github.io/django-components/latest/concepts/advanced/rendering_js_css).

Do you have the middleware configured?

---

> Also I'd like to know if there's a way to pass context variables to the css, i.e allow something like        max-width: {{ max_with }} !important; for the css of the component.

It's still a work in progress, but soon you'll be able to pass variables to CSS using `get_css_data()`:
```py
class MyTable(Component):
    def get_css_data(self, max_width: int, **kwargs):
        return {
            "max-width": f"{max_width}px",
        }

    css = """
      .my-table {
        max-width: var(--max-width),
      }
    """
```

I'm finishing up [scoped CSS](https://vue-loader.vuejs.org/guide/scoped-css.html) today, and the next feature in line will be CSS variables. But there's a few more PRs before that, so it could be out in maybe 2-3 weeks.

**Reply by spapas** [reply]
Hello @JuroOravec holidays are (still) great, hope the same for you :)

I've completely missed the docs page you mentioned; it works great now thank you!

The css variables solution seems very good so i'll definitely use it when it's available.

Thank you and kind regards,
Serafeim
